lib/Doctrine/ORM/Mapping/DefaultQuoteStrategy.php

```diff
@@ -45,9 +45,19 @@ public function getColumnName($fieldName, ClassMetadata $class, AbstractPlatform
      */
     public function getTableName(ClassMetadata $class, AbstractPlatform $platform)
     {
-        return isset($class->table['quoted']) 
-            ? $platform->quoteIdentifier($class->table['name'])
-            : $class->table['name'];
+        $tableName = $class->table['name'];
+
+        if ( ! empty($class->table['schema'])) {
+            $tableName = $class->table['schema'] . '.' . $class->table['name'];
+
+            if ( ! $platform->supportsSchemas() && $platform->canEmulateSchemas()) {
+                $tableName = $class->table['schema'] . '__' . $class->table['name'];
```

Comment:
> I actually pulled all changes locally for merge, so expect it to be merged in the next hours

Great! Thank you! I should really consider making some other contributions to Doctrine.